x/staking/autocli.go

```diff
@@ -0,0 +1,140 @@
+					PositionalArgs: []*autocliv1.PositionalArgDescriptor{
+						{ProtoField: "validator_addr"},
+					},
+				},
+				{
+					RpcMethod: "Delegation",
+					Use:       "delegation [delegator-addr] [validator-addr]",
+					Short:     "Query a delegation based on address and validator address",
+					Example: fmt.Sprintf(`%s query staking delegation [delegator-address] [validator-address]`,
+						version.AppName),
+					Long: "Query delegations for an individual delegator on an individual validator",
+					PositionalArgs: []*autocliv1.PositionalArgDescriptor{
+						{ProtoField: "delegator_addr"},
+						{ProtoField: "validator_addr"},
+					},
+				},
+				{
+					RpcMethod: "UnbondingDelegation",
+					Use:       "unbonding-delegation [delegator-addr] [validator-addr]",
+					Short:     "Query an unbonding-delegation record based on delegator and validator address",
+					Long:      "Query unbonding delegations for an individual delegator on an individual validator.",
+					Example:   fmt.Sprintf("$ %s query staking unbonding-delegation [delegator-adr] [validator-adr]", version.AppName),
+					PositionalArgs: []*autocliv1.PositionalArgDescriptor{
+						{ProtoField: "delegator_addr"},
+						{ProtoField: "validator_addr"},
+					},
+				},
+				{
+					RpcMethod: "DelegatorDelegations",
+					Use:       "delegations [delegator-addr]",
+					Short:     "Query all delegations made by one delegator",
+					Long:      "Query delegations for an individual delegator on all validators.",
+					Example:   fmt.Sprintf("$ %s query staking delegations [validator-addr]", version.AppName),
+					PositionalArgs: []*autocliv1.PositionalArgDescriptor{
+						{ProtoField: "delegator_addr"},
+					},
+				},
+				{
+					RpcMethod: "DelegatorUnbondingDelegations",
+					Use:       "unbonding-delegations [delegator-addr]",
+					Short:     "Query all unbonding-delegations records for one delegator",
+					Long:      "Query unbonding delegations for an individual delegator.",
+					Example:   fmt.Sprintf("$ %s query staking unbonding-delegations [delegator-addr]", version.AppName),
+					PositionalArgs: []*autocliv1.PositionalArgDescriptor{
+						{ProtoField: "delegator_addr"},
+					},
+				},
+				{
+					RpcMethod: "Redelegations",
+					Use:       "redelegation [delegator-addr] [src-validator-addr] [dst-validator-addr]",
+					Short:     "Query a redelegation record based on delegator and a source and destination validator address",
+					Long:      "Query a redelegation record for an individual delegator between a source and destination validator.",
+					Example:   fmt.Sprintf("$ %s query staking redelegation [delegator-addr] [src-validator-addr] [dst-validator-addr]", version.AppName),
+					PositionalArgs: []*autocliv1.PositionalArgDescriptor{
+						{ProtoField: "delegator_addr"},
+						{ProtoField: "src_validator_addr"},
+						{ProtoField: "dst_validator_addr"},
+					},
+				},
+				{
+					RpcMethod: "HistoricalInfo",
```

Comment:
Getting `6:33PM ERR failure when running app err="proto: google.protobuf.Any: unable to resolve \"/cosmos.crypto.ed25519.PubKey\": not found"`

Reply:
btw, I am looking at this myself for auth

Reply:
solved with https://github.com/cosmos/cosmos-sdk/pull/16641/commits/953351f096bcd0385d7a4506dfa713744b478c40